src/hooks/useSafeNotifications.ts

```diff
@@ -37,6 +42,44 @@ const useSafeNotifications = (): void => {
       dispatch(closeNotification({ id }))
     }
   }, [dispatch, chainId, safeAddress, implementationVersionState, version, chain?.shortName])
+
+  /**
+   * Show a notification when the Safe master copy is not supported
+   */
+
+  const masterCopy = safe.implementation.value
+
+  const [validMasterCopy] = useAsync(async () => {
+    if (masterCopy) {
+      return await isValidMasterCopy(chainId, masterCopy)
+    }
+  }, [chainId, masterCopy])
+
+  useEffect(() => {
+    if (validMasterCopy === undefined || validMasterCopy) {
+      return
+    }
```

Comment:
I suggest to use the `loading` flag from `useAsync` here instead of checking if the return value is undefined

Reply:
The `loading` flag is not reliabl:

![loading flag](https://user-images.githubusercontent.com/20442784/191962931-17f46164-a29c-476e-844e-7cd10a7b0a67.gif)

I even tried moving `setLoading` to the top of the `useEffect` to no avail. I think it's related to React 18's batched updates but `flushSync` didn't help either.

It's something we should look into at some point as I agree that it is a better flag to reference.